### ISSUE TITLE
PD010: pivot table false positive

### ISSUE THREAD

**Issue by sbrugman** [charliermarsh/ruff#2143]
When a Spark DataFrame is used, PD010 will still generate an error:

```
dataframe.groupBy("my_column")
  .pivot("other_column")
```

> PD010 `.pivot_table` is preferred to `.pivot` or `.unstack`; provides same functionality

### Suggested solution
When type inference is too complicated, do not consider this rule when `pyspark` is in the imports.

**Comment by leuchtum** [comment]
Another false positive, as `pd.Series` has no method `pivot_table`. The method only exist on `pd.DataFrames`.

Example:
```python
import pandas as pd

idx1 = [1, 2, 3, 4, 5]
idx2 = [10, 20, 30, 40, 50]
multi_idx = pd.MultiIndex.from_arrays([idx1, idx2])
s = pd.Series(data=[100, 200, 300, 400, 500], index=multi_idx)

unstacked = s.unstack()
```

Output:
```shell
$ ruff demo_ruff_PD010.py --select=PD010
demo_ruff_PD010.py:8:13: PD010 `.pivot_table` is preferred to `.pivot` or `.unstack`; provides same functionality
Found 1 error.
```

**Comment by janosh** [comment]
here's a slightly simpler repro for multiple false-positive PD010 on `pd.Series`

```py
import pandas as pd

df = pd.DataFrame({"A": [1, 2, 3]})
df["A"].unstack()  # noqa: PD010
df.A.unstack()  # noqa: PD010

pd.Series([1,2,3]).unstack()  # noqa: PD010
```

**Comment by anor4k** [comment]
Also get a false positive on polars DataFrames.
